script.js

```diff
@@ -1,4 +1,91 @@
+//global variables
+var currentGameMode = `waiting for username`;
+var userName = ``;
+//Number Games of Won
+var numGamesPlayed = 0;
+console.log(numGamesPlayed);
+//Number of Games Played
+var numGamesWon = 0;
+console.log(numGamesWon);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+  if (currentGameMode == `waiting for username`) {
+    //set the name
+    userName = input;
+    console.log(userName);
+    currentGameMode = `SPSGame`;
+
+    myOutputValue = `Hello ${userName}`;
+  } else if (currentGameMode == `SPSGame`) {
+    //Increase number of Games Played
+    numGamesPlayed += 1;
+
+    //SPS Generator
+    var SPSArray = ["Scissors", "Paper", "Stone"];
+    var randomSPS = Math.floor(Math.random() * SPSArray.length); //RandomSPS
+    console.log(SPSArray[randomSPS]);
+
+    //Player Beat Computer Function
+    var didPlayerBeatComputer = function (input) {
+      return (
+        (input == "Scissors" && SPSArray[randomSPS] == "Paper") ||
+        (input == "Paper" && SPSArray[randomSPS] == "Stone") ||
+        (input == "Stone" && SPSArray[randomSPS] == "Scissors") ||
+        (input == "ReversedScissors" && SPSArray[randomSPS] == "Stone") ||
+        (input == "ReversedStone" && SPSArray[randomSPS] == "Paper") ||
+        (input == "ReversedPaper" && SPSArray[randomSPS] == "Scissors")
+      );
+    };
```

Comment:
this function should be placed outside the main function to keep the main clean
SPSArray[randomSPS] can be assigned to a variable so that its easier to read and type

var computerChoice = SPSArray[randomSPS];
(input == "Scissors" && computerChoice == "Paper")

you may have decided to put this function in the main because it won't work otherwise. to make it workable, you can put the randomNum generator (line27) in a helper function to be called in the didPlayerBeatComputer function to give you randomSPS.

Good shot at using arrays and referencing that.